.travis/deploy-lib.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+cd `dirname $0`/..
+
+if [ -z "${SONATYPE_USERNAME}" ]
+then
+    echo "error: please set SONATYPE_USERNAME and SONATYPE_PASSWORD environment variable"
+    exit 1
+fi
+
+if [ -z "${SONATYPE_PASSWORD}" ]
+then
+    echo "error: please set SONATYPE_PASSWORD environment variable"
+    exit 1
+fi
+
+if [ ! -z "${TRAVIS_TAG}" ]
+then
+    echo "on a tag -> set pom.xml <version> to ${TRAVIS_TAG}"
+    mvn --settings .travis/settings.xml org.codehaus.mojo:versions-maven-plugin:2.1:set -DgenerateBackupPoms=false -DnewVersion=${TRAVIS_TAG} 1>/dev/null 2>/dev/null
+    mvn --batch-mode --settings .travis/settings.xml --update-snapshots clean deploy -Ddeploy -DskipTests
+
+    # Create new SNAPSHOT version by incrementing the patch
+    TRAVIS_TAG_ARRAY=( ${TRAVIS_TAG//./ } )
+    ((TRAVIS_TAG_ARRAY[-1]++))
+
+    function join { local IFS="$1"; shift; echo "$*"; }
+    NEW_SNAPSHOT_VERSION="$(join . ${TRAVIS_TAG_ARRAY[@]})-SNAPSHOT"
```

Comment:
Similar to the deploy script from midas, except more generalized. This will just grab the last part of the version, increment it, and create a new snapshot version. For example:

On parent-pom -  current version: 2-SNAPSHOT > git tag 2 HEAD > new version: 3-SNAPSHOT
On midas - current version: 1.0.2-SNAPSHOT > git tag 2.0.0 HEAD > new version 2.0.1-SNAPSHOT